src/custom/components/AccountDetails/Transaction/index.tsx

```diff
@@ -382,21 +382,28 @@ export default function Transaction({ hash: id }: { hash: string }) {
 
         <StatusLabelWrapper>
           <StatusLabel color={determinePillColour(status, type)} isPending={isPending} isCancelling={isCancelling}>
-            {isConfirmed ? (
+            {isConfirmed && isTransaction ? (
+              <SVG src={OrderCheckImage} description="Transaction Confirmed" />
+            ) : isConfirmed ? (
               <SVG src={OrderCheckImage} description="Order Filled" />
+            ) : isExpired && isTransaction ? (
+              <SVG src={OrderCancelledImage} description="Transaction Failed" />
             ) : isExpired ? (
               <SVG src={OrderExpiredImage} description="Order Expired" />
             ) : isCancelled ? (
               <SVG src={OrderCancelledImage} description="Order Cancelled" />
             ) : isCancelling ? null : (
               <SVG src={OrderOpenImage} description="Order Open" />
             )}
+
             {isPending
               ? 'Open'
               : isConfirmed && isTransaction
               ? 'Approved'
               : isConfirmed
               ? 'Filled'
+              : isExpired && isTransaction
```

Comment:
nitpicking: there's a bit of logic repetition. In principle, you can do a pure javascript function, which receives the `ActivityType` and the `ActivityStatus` and they map it to a new type with all statuses. This new type could even be used to get the values from a static`STATUS_DETAILS` constant, so each state has it's label/image/description

```jsx
const statusUi = getStatusUi(status, type) // not sure how to call this, probably some of u come up with a better name
const { image, label, description } = STATUS_DETAILS[statusUi]

<SVG src={image} description={description} /> {label}
```